Userland/Libraries/LibCore/DirIterator.cpp

```diff
@@ -60,6 +61,13 @@ bool DirIterator::advance_next()
         m_next = DirectoryEntry::from_stat(m_dir, *de);
 #else
         m_next = DirectoryEntry::from_dirent(*de);
+
+        // dirent structures from readdir aren't guaranteed to contain valid file types,
+        // as it is possible that the underlying filesystem doesn't keep track of those.
+        if (m_next->type == DirectoryEntry::Type::Unknown) {
+            auto stat = MUST(Core::System::fstat(dirfd(m_dir)));
+            m_next->type = DirectoryEntry::directory_entry_type_from_stat(stat.st_mode);
```

Comment:
:thinking: is this really a sane assumption to make? why should the child's type be the same as the parent's type?
or am I reading this wrong?